### PR TITLE
[BIOMAGE-1793] - Connect to staging RDS deployment

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
####  Problem
We want to have a separate staging RDS deployment so we can test changes to the database in isolation.

#### Solution
[IAC PR 326](https://github.com/hms-dbmi-cellenics/iac/pull/3266) added sandbox id to deployed RDS instances. The changes in this PR : 
- Adds a new flag `--with_rds` that can be used with the `biomage stage` command to decide whether we want to stage a new RDS instance with the staging environment or not. If no staging environment is created, the API will connect to the `default` RDS instance.
- Add a new parameter `sandbox_id` to the `biomage rds tunnel` command so that the RDS agent can tunnel to the correct RDS instance.

#### Link to issue
https://biomage.atlassian.net/browse/BIOMAGE-1793

#### Link to staging deployment URL 
N/A

#### Links to any Pull Requests related to this

- IAC - https://github.com/hms-dbmi-cellenics/iac/pull/326
- API - https://github.com/hms-dbmi-cellenics/api/pull/323
- Biomage utils - https://github.com/hms-dbmi-cellenics/biomage-utils/pull/81

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] New commands have been added to Makefile/test
- [ ] Unit tests written
- [ ] Run make test
- [X] Test any modified command

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: biomage/stage/stage.py [hunk 1]
```diff
@@ -206,12 +206,20 @@ def get_sandbox_id(templates, manifests, all_experiments, auto):
...
     if auto:
-        return (user_name + "-" + random_name.generate_name(
-            lists=(
-                random_name.dictionaries.ADJECTIVES,
-                random_name.dictionaries.ANIMALS,
+        return (
+            (
+                user_name
+                + "-"
+                + random_name.generate_name(
```
Comment (cosa65) [inline biomage/stage/stage.py:213]: Was this done by the linter? I feel like it was more readable without everything so split up

File: biomage/stage/stage.py [hunk 2]
```diff
@@ -305,6 +313,22 @@ def create_manifest(templates, token, repo_to_ref, all_experiments, auto):
         click.echo("\n".join(f"• {pin}" for pin in pins))
     else:
         click.echo("Not pinning any repository")
+
+    if not auto:
```
Comment (cosa65) [inline biomage/stage/stage.py:317]: What is auto?

File: .gitignore [hunk 3]
```diff
@@ -141,4 +141,4 @@ cython_debug/
 
 .vscode
 .DS_Store
-data/
+data/
```
Comment (ivababukova) [inline .gitignore:144]: do we use this folder at all?

File: biomage/stage/stage.py (hunk 2 above)
Comment (aerlaut) [reply biomage/stage/stage.py:317]: it's a settings for the `biomage stage` command. If you do `biomage stage ... --auto`, it'll automatically assign a random name for your deployment, and confirm "Yes" for the options.

File: .gitignore (hunk 3 above)
Comment (aerlaut) [reply .gitignore:144]: Yes. it's default directory for `biomage experiment pull`

**Review by cosa65 — APPROVED** [review]

**Pull request merged**